contracts/stake-cw721/src/contract.rs

```diff
@@ -0,0 +1,1844 @@
+                    .add_attribute("token_id", token_id)
+                    .add_attribute("claim_duration", format!("{}", duration)));
+            }
+
+            let outstanding_reward_claims = REWARD_CLAIMS
+                .query_claims(deps.as_ref(), &info.sender)?
+                .claims;
+            if outstanding_reward_claims.len() >= MAX_CLAIMS as usize {
+                return Err(ContractError::TooManyClaims {});
+            }
+
+            REWARD_CLAIMS.create_claim(
+                deps.storage,
+                &validated_reward_wallet
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ADDR.clone()),
+                amount_to_claim,
+                duration.after(&env.block),
+            )?;
+
+            Ok(Response::new()
+                .add_attribute("action", "unstake")
+                .add_submessages(hook_msgs)
+                .add_attribute("from", info.sender)
+                .add_attribute("token_id", token_id)
+                .add_attribute("claim_duration", format!("{}", duration))
+                .add_attribute(
+                    "reward_wallet",
+                    validated_reward_wallet.unwrap_or_else(|| DEFAULT_ADDR.clone()),
+                )
+                .add_attribute("reward_amount", amount_to_claim))
+        }
+    }
+}
+
+pub fn execute_claim_nfts(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let nfts = NFT_CLAIMS.claim_nfts(deps.storage, &info.sender, &_env.block)?;
+    if nfts.is_empty() {
+        return Err(ContractError::NothingToClaim {});
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    let mut msgs = Vec::new();
+    for nft in nfts.iter() {
+        let cw_transfer_msg = cw721::Cw721ExecuteMsg::TransferNft {
+            recipient: info.sender.to_string(),
+            token_id: nft.clone(),
+        };
+
+        let wasm_msg = cosmwasm_std::WasmMsg::Execute {
+            contract_addr: config.nft_address.to_string(),
+            msg: to_binary(&cw_transfer_msg)?,
+            funds: vec![],
+        };
+        msgs.push(wasm_msg);
+    }
+
+    Ok(Response::new()
+        .add_messages(msgs)
+        .add_attribute("action", "claim_nfts")
+        .add_attribute("from", info.sender.clone()))
+}
+
+pub fn execute_claim_rewards(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let reward_amount =
+        REWARD_CLAIMS.claim_tokens(deps.storage, &info.sender, &_env.block, None)?;
+    if reward_amount.is_zero() {
+        return Err(ContractError::NothingToClaim {});
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    let cw_send_msg = cw20::Cw20ExecuteMsg::Transfer {
+        recipient: info.sender.to_string(),
+        amount: reward_amount,
+    };
+
+    let wasm_msg = cosmwasm_std::WasmMsg::Execute {
+        contract_addr: config.reward_token_address.unwrap().to_string(),
```

Comment:
This unwrap here seems liable to cause trouble. Could we return an error if this is `None` instead? Understand that it may be the case currently that we will never hit this condition if there is no reward token.

Reply:
Cool.